experiment_code/hackrl/experiment.py

```diff
@@ -35,18 +39,106 @@
 from hackrl.core import nest
 from hackrl.core import record
 from hackrl.core import vtrace
-
 # TTYREC_ASYNC_ITERATOR = None
 # TTYREC_DATA = None
 TTYREC_HIDDEN_STATE = None
 TTYREC_ENVPOOL = None
+EWC_INSTANCE = None
+
+
+class EWC(object):
+    def __init__(self, model: nn.Module, data: dict):
+
+        self.model = model
+        self.data = data
+
+        env_outputs = data["env_outputs"]
+        initial_core_state = data["initial_core_state"]
+        self.dataset = [(env_outputs, initial_core_state)]
+
+        self.params = {
+            n: p
+            for n, p in self.model.named_parameters()
+            if p.requires_grad and "baseline".casefold() not in n.casefold()
+        }
+        self._means = {}
+        self._precision_matrices = self._diag_fisher()
+
+        for n, p in copy.deepcopy(self.params).items():
+            self._means[n] = p.data
+
+    def _diag_fisher(self):
+        precision_matrices = {}
+        for n, p in deepcopy(self.params).items():
+            p.data.zero_()
+            precision_matrices[n] = p.data
+
+        self.model.eval()
+        cudnn.enabled = False
```

Comment:
I don't follow why we need this line. Can you please explain?

Reply:
Without this line there is an Error with backpropagation in eval mode for model with RNN